nubison_model/Service.py

```diff
@@ -37,6 +42,20 @@ def load_nubison_model(
     return nubison_model
 
 
+@contextmanager
+def test_client(model_uri):
+    app = build_inference_service(mlflow_model_uri=model_uri)
+    # Disable metrics for testing. Avoids Prometheus client duplicated registration error
+    app.config["metrics"] = {"enabled": False}
+
+    # Create a temporary directory and set it as the current working directory to run tests
+    # To avoid model initialization conflicts with the current directory
+    test_dir = TemporaryDirectory()
+    with temporary_cwd(test_dir.name), TestClient(app.to_asgi()) as client:
+
+        yield client
+
+
 def build_inference_service(
     mlflow_tracking_uri: Optional[str] = None, mlflow_model_uri: Optional[str] = None
 ):
```

Comment:
I was curious about the type hint. Optional[str] means that the variable can be of type str or None, but it seems like this variable won’t work properly if its value is None. Wouldn’t it be more appropriate to set a default URI instead of allowing None?

Reply:
You can read initializing parameters right below the function definition.
Initializing in function body has several advantages over doing it in function definition like `mlflow_tracking_uri: str = DEFAULT_URI`

In this case those values should be initialized in priority below.
given param > env var > default value

It can't be done by param default value.

Reply:
Allowing `None` to param value has several meanings too.
One of those is to sure no value is passed to the param.

And if you miss the default value `None` with `Optional[str]` type,
callers should provide the value even if the value is `None`
`uri: Optional[str] = None` => `func()`
vs
`uri: Optional[str]` => `func(None)`

Reply:
I understand, thank you for your answer.